Mage/src/main/java/mage/abilities/common/CaseAbility.java

```diff
@@ -0,0 +1,128 @@
+package mage.abilities.common;
+
+import mage.abilities.Ability;
+import mage.abilities.condition.CompoundCondition;
+import mage.abilities.condition.Condition;
+import mage.abilities.condition.common.SolvedSourceCondition;
+import mage.abilities.effects.OneShotEffect;
+import mage.constants.Outcome;
+import mage.constants.TargetController;
+import mage.constants.Zone;
+import mage.game.Game;
+import mage.game.permanent.Permanent;
+import mage.util.CardUtil;
+
+/**
+ * The Case mechanic was added in Murders at Karlov Manor [MKM].
+ * <ul>
+ * <li>Each Case has two special keyword abilities: to solve and solved.</li>
+ * <li>"To Solve — [condition]" means "At the beginning of your end step,
+ * if [condition] and this Case is not solved, it becomes solved."</li>
+ * <li>The meaning of "solved" differs based on what type of ability follows it.
+ * "Solved — [activated ability]" means "[Activated ability].
+ * Activate only if this Case is solved." Activated abilities contain a colon.
+ * They're generally written "[Cost]: [Effect]."</li>
+ * <li>"Solved — [Triggered ability]" means "[Triggered ability].
+ * This ability triggers only if this Case is solved."
+ * Triggered abilities use the word "when," "whenever," or "at."
+ * They're often written as "[Trigger condition], [effect]."</li>
+ * <li>"Solved — [static ability]" means "As long as this Case is solved, [static ability]."
+ * Static abilities are written as statements, such as "Creatures you control get +1/+1"
+ * or "Instant and sorcery spells you cast cost {1} less to cast."</li>
+ * <li>"To solve" abilities will check for their condition twice:
+ * once when the ability would trigger, and once when it resolves.
+ * If the condition isn't true at the beginning of your end step,
+ * the ability won't trigger at all.
+ * If the condition isn't true when the ability resolves, the Case won't become solved.</li>
+ * <li>Once a Case becomes solved, it stays solved until it leaves the battlefield.</li>
+ * <li>Cases don't lose their other abilities when they become solved.</li>
+ * <li>Being solved is not part of a permanent's copiable values.
+ * A permanent that becomes a copy of a solved Case is not solved.
+ * A solved Case that somehow becomes a copy of a different Case stays solved.</li>
+ * </ul>
+ *
+ * @author DominionSpy
+ */
+public class CaseAbility extends SimpleStaticAbility {
+
+    public CaseAbility(Ability initialAbility, Condition toSolveCondition, Ability solvedAbility) {
```

Comment:
To aid future devs, could add a doc comment for this constructor clarifying which conditional classes are appropriate to use (and that they use `SolvedSourceCondition.SOLVED`, `hideCondition()` etc.)

Reply:
Re: `hideCondition()`: I've moved this into the `CaseAbility` class, since it only needs to be added to `ConditionalActivatedAbility` and this is now being checked.